smartapps/cuchen-wifi-cooker/cuchen-wifi-cooker.src/cuchen-wifi-cooker.groovy

```diff
@@ -0,0 +1,242 @@
+/**
+ *  Cuchen Wifi Cooker
+ *
+ *  Copyright 2015 Cuchen
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at:
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ *
+ */
+ 
+import groovy.json.*
+import static java.util.Calendar.*
+
+definition(
+    name: "Cuchen Wifi Cooker",
+    namespace: "cuchen.wifi.cooker",
+    author: "Cuchen",
+    description: "control your Cuchen Wifi Cooker.",
+    category: "My Apps",
+    iconUrl: "http://lihom.jc-square.com/cuchen.png",
+    iconX2Url: "http://lihom.jc-square.com/cuchen@2x.png"
+    //iconX3Url: "https://s3.amazonaws.com/smartapp-icons/Convenience/Cat-Convenience@2x.png"
+    )
+
+
+preferences {
+	def msg = """Tap 'Next' after you have entered in your Cooker credentials.
+
+Once your credentials are accepted, SmartThings will scan your installation for Cooker."""
+
+	page(name: "selectDevices", title: "Connect Your Cuchen Wifi Cooker to SmartThings", install: true, uninstall: true) {
+		section("Connected Cooker Credentials") {
+			input "cookerId", "text", title: "Enter Cuchen Cooker Id(Mac address)", required: true
+			input "password", "password", title: "Enter Cooker Password", required: true
+			paragraph msg
+		}
+	}
+
+	page(name: "chooseCooker", title: "Choose Cooker to Control With SmartThings", content: "initialize")
+}
+
+def installed() {
+	log.debug "Installed with settings: ${settings}"
+
+	initialize()
+}
+
+def updated() {
+	log.debug "Updated with settings: ${settings}"
+    
+	initialize()
+}
+
+def uninstalled() {
+	def preInstalled = getChildDevice(settings.cookerId)
+    if (preInstalled) {
+	    deleteChildDevice(settings.cookerId)
+    }
+}
+
+def initialize() {
+	def preInstalled = getChildDevice(settings.cookerId)
+    if (preInstalled) {
+	    deleteChildDevice(settings.cookerId)
+    }
+    
+	// TODO: subscribe to attributes, devices, locations, etc.
+    def dev = addChildDevice("cuchen.wifi.cooker.pa-10", "Cuchen Wifi Cooker PA-10", settings.cookerId, null, [cookerId : settings.cookerId, password: settings.password, compeletedSetup: true])
+    log.debug "Device commands = " + dev.supportedCommands
+    log.debug "Device attrs = " + dev.supportedAttributes
+    dev.refreshStatus()
+}
+
+// TODO: implement event handlers
+
+def poll(childDevice) {
+  log.debug "Executing 'poll'"
+  getStatus(settings.cookerId, true)
+}
+
+def refreshStatus() {
+  def dev = getChildDevice(settings.deviceId)
```

Comment:
that is my fault. "getChildDevice(settings.cookerId)" is right.

"cookerId" is preferences.
"deviceId" is "cookerId" for, API param name of our server.